libmamba/src/core/util.cpp

```diff
@@ -1133,7 +1137,7 @@ namespace mamba
         try
         {
             // Don't even log if the file/directory isn't writable by someone or doesnt exists.
-            if (path::is_writable(path))
+            if (Context::instance().disable_lockfile || path::is_writable(path))
```

Comment:
Either I lack context or this is wrong, shouldn't it be `!disable_lockfile && is_writable`? If so, this lacks a test.

Reply:
To revert this change you would need to modify this function:
https://github.com/danpf/mamba/blob/danpf-nolock/micromamba/src/run.cpp#L105 to not fail if nullptr is returned.

I also think it should not matter if `path::is_writable` or not if we are not making a lockfile anyway so I think this is correct.

Reply:
I think the logic is now confusing (although I think I get it). Maybe we can return earlier when the lockfile is disable? Or do we even need that check here?